### PR TITLE
Higher-order symmetric quadrature rules on the triangle and tetrahedron

### PR DESCRIPTION
**Pull request by tristanmontoya**
I added Xiao-Gimbutas quadrature rules of degree 28-50 on the triangle (previously only rules of up to degree 27 were included), and symmetric quadrature rules on the tetrahedron recently introduced by [Jaśkowiec and Sukumar](https://onlinelibrary.wiley.com/doi/abs/10.1002/nme.6528) for degrees up to 20.

I made the new quadrature rules the defaults for degrees 28-50 on the triangle and 16-20 on the tetrahedron, where previously a Stroud-type quadrature would be used. As such, the test which was meant to test Stroud quadrature now chooses one of the newly added symmetric quadrature rules, so I added a new test of even higher order to make sure that the Stroud rules still got tested.

The stroud_quad_nodes and jaskowiek_sukumar_quad_nodes functions are also now exported to give users more control over the quadrature being used.

Hopefully I did this correctly this time... let me know if there are any issues with this PR, I'm still pretty new to contributing to other peoples' projects on GitHub.

### CODE REVIEW

File: src/tet_element.jl [hunk 1]
```diff
@@ -99,6 +99,8 @@ function quad_nodes_tet(N)
         s = rstw[:,2]
         t = rstw[:,3]
         w = rstw[:,4]
+    elseif N<21
```
Comment (jlchan) [inline src/tet_element.jl]:
```suggestion
    elseif N < 21
```

File: src/tet_element.jl [hunk 2]
```diff
@@ -99,6 +99,8 @@ function quad_nodes_tet(N)
...
+        r,s,t,w = jaskowiec_sukumar_quad_nodes(Tet(),N)
```
Comment (jlchan) [inline src/tet_element.jl]:
```suggestion
        r, s, t, w = jaskowiec_sukumar_quad_nodes(Tet(), N)
```

File: src/tet_element.jl [hunk 3]
```diff
@@ -107,6 +109,23 @@ function quad_nodes_tet(N)
...
+Jaśkowiec, J, Sukumar, N., "High-order symmetric cubature rules for tetrahedra and pyramids." Int J Numer Methods Eng. 122(1): 148-171, 2021.
+"""
+function jaskowiec_sukumar_quad_nodes(elem::Tet,N)
+    if N > 20
+        @error "Jaskowiek-Sukumar quadrature rules not available for N > 20."
+    end
+
+    rstw::Matrix{Float64} = readdlm(string(@__DIR__,"/QuadratureData/jaskowiec_sukumar_nodes_tet_N", N, ".txt"),' ', Float64, '\n')
+
+    return rstw[:,1],rstw[:,2],rstw[:,3],rstw[:,4]
```
Comment (jlchan) [inline src/tet_element.jl]:
```suggestion
function jaskowiec_sukumar_quad_nodes(elem::Tet, N)
    if N > 20
        @error "Jaskowiek-Sukumar quadrature rules not available for N > 20."
    end

    rstw::Matrix{Float64} = readdlm(string(@__DIR__,"/QuadratureData/jaskowiec_sukumar_nodes_tet_N", N, ".txt"), ' ', Float64, '\n')

    return rstw[:,1], rstw[:,2], rstw[:,3], rstw[:,4]
```

**Review by jlchan — CHANGES_REQUESTED** [review]
This looks amazing - thanks @tristanmontoya! I didn't know these existed, so that's pretty fantastic. 

I added some very minor formatting suggestions. In case it helps, you can commit these by just clicking on "Commit suggestion" on the Github page for this pull request.

**Comment by jlchan** [comment]
I went ahead and added the minor formatting commits - once the tests pass this should be good to merge!

**Comment by tristanmontoya** [comment]
Awesome, thanks Jesse! Glad to be able to contribute.

**Pull request merged**